src/main/java/querqy/elasticsearch/aggregation/InternalDecorationAggregation.java

```diff
@@ -0,0 +1,127 @@
+package querqy.elasticsearch.aggregation;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.metrics.ScriptedMetric;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.singletonList;
+
+public class InternalDecorationAggregation extends InternalAggregation implements ScriptedMetric {
+
+    private final List<Object> aggregations;
+
+    InternalDecorationAggregation(final String name, final List<Object> aggregations, final Map<String, Object> metadata) {
+        super(name, metadata);
+        this.aggregations = aggregations;
+    }
+
+    public InternalDecorationAggregation(final StreamInput in) throws IOException {
+        super(in);
+        if (in.getVersion().before(Version.V_7_8_0)) {
+            aggregations = singletonList(in.readGenericValue());
+        } else {
+            aggregations = in.readList(StreamInput::readGenericValue);
+        }
+    }
+
+    @Override
+    protected void doWriteTo(final StreamOutput out) throws IOException {
+        if (out.getVersion().before(Version.V_7_8_0)) {
```

Comment:
I don't know if we need this as this Querqy version doesn't support ES 7.8.0 and lower at all. Do we really need this check?

Reply:
If we're not going to downgrade, we don't need it...
You prefer to have it removed?

Reply:
Yes please!

Reply:
done